docs/en/operations/server-configuration-parameters/settings.md

```diff
@@ -1568,6 +1570,28 @@ The log level of individual log names can be overridden. For example, to mute al
 </logger>
 ```
 
+**Regular Expression Filtering**
+
+The messages logged can be filtered using regular expressions using `message_regexp` and `message_regexp_negative`. This can be done on a per-level basis or globally. If both are specified for a particular logger, the global expression is ignored and the per-level one overrides it.
+
+
+```xml
+<logger>
+    <!-- Global: Only log messages that have 'executeQuery' in them and not 'ConfigReloader' -->
+    <message_regexp>.*executeQuery.*</message_regexp>
+    <message_regexp>.*ConfigReloader.*</message_regexp>
+
+    <levels>
+        <logger>
+            <name>RBAC</name>
+            <!-- For logger 'RBAC', instead of matching for '.*executeQuery.*' and '.*ConfigReloader.*' match instead for '.*Application.*' and '.*Setting.*'. -->
+            <message_regexp>.*Application.*</message_regexp>
+            <message_regexp_negative>.*Setting.*</message_regexp_negative>
+        </logger>
+    </levels>
```

Comment:
Also, this shouldn't be inside the `level` tag. Logging level is `trace`, `info`, `debug`, `error`, etc. This is not connected to levels.

What do you say about

```
<logger>
    <!-- Global: Only log messages that have 'executeQuery' in them and not 'ConfigReloader' -->
    <message_regexp>.*executeQuery.*</message_regexp>
    <message_regexp>.*ConfigReloader.*</message_regexp>

    <message_regexps>
        <logger>
            <name>RBAC</name>
            <!-- For logger 'RBAC', instead of matching for '.*executeQuery.*' and '.*ConfigReloader.*' match instead for '.*Application.*' and '.*Setting.*'. -->
            <message_regexp>.*Application.*</message_regexp>
            <message_regexp_negative>.*Setting.*</message_regexp_negative>
        </logger>
    </message_regexps>
</logger>
```

Reply:
Sure, just changed the behavior to this.